contracts/PegStability/PegStability.sol

```diff
@@ -347,4 +286,66 @@ contract PegStability is AccessControlledV8, ReentrancyGuardUpgradeable {
         comptroller = comptroller_;
         emit ComptrollerChanged(oldComptrollerAddress, comptroller_);
     }
+
+    /*** Helper Functions ***/
+
+    /**
+     * @dev Calculates the USD value of the given amount of stable tokens depending on the swap direction.
+     * @param amount The amount of stable tokens.
+     * @param direction The direction of the swap.
+     * @return The USD value of the given amount of stable tokens scaled by 1e18 taking into account the direction of the swap
+     */
+    function previewTokenUSDAmount(uint256 amount, FeeDirection direction) internal view returns (uint256) {
+        return (amount * getPriceInUSD(direction)) / MANTISSA_ONE;
+    }
+
+    /**
+     * @notice Get the price of stable token in USD, based on the selected oracle
+     * @dev This function returns either min(1$,oraclePrice) or max(1$,oraclePrice) depending on the direction of the swap
+     * @param direction The direction of the swap: FeeDirection.IN or FeeDirection.OUT
+     * @return The price in USD, adjusted based on the selected direction
+     */
+    function getPriceInUSD(FeeDirection direction) internal view returns (uint256) {
+        address priceOracleAddress = OracleProviderInterface(comptroller).oracle();
+        uint256 price = IPriceOracle(priceOracleAddress).getUnderlyingPrice(vTokenAddress);
+        require(price != INVALID_ORACLE_PRICE, "Invalid oracle price.");
+        if (direction == FeeDirection.IN) {
+            //MIN (1,price)
+            return MANTISSA_ONE < price ? MANTISSA_ONE : price;
```

Comment:
We have to scale the `price` before comparing it with `MANTISSA_ONE`, to have it with 18 decimals. The price returned by the Oracle is defined with (36 - asset decimals) decimals. So, I suppose we can divide `price` by `10 ** (18 - asset decimals)`

Moreover, we shouldn't return directly `MANTISSA_ONE`. We should scale it to have the expected decimals (36 - asset decimals)